### PR TITLE
Issue/82 coverage

### PR DESCRIPTION
**Pull request by vitalych**
https://github.com/S2E/s2e-env/issues/82

### CODE REVIEW

File: s2e_env/commands/code_coverage/line_info/dwarf.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                    src_path = os.path.realpath(src_path)
+                    file_line_info[src_path] = {}
+
+                file_line_info[src_path][state.line] = addr_counts.get(state.address, 0)
+
+    return file_line_info
+
+
+def get_file_line_coverage(target_path, addr_counts):
+
```
Comment (adrianherrera) [inline s2e_env/commands/code_coverage/line_info/dwarf.py]: Delete this line

File: s2e_env/commands/code_coverage/line_info/__init__.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                    'path/to/source': {
+                                          1: 2, # Line 1 was executed twice
+                                          2: 5, # Line 2 was executed five times
+                                          ...
+                                      },
+                    ...
+                }
+                ```
+    """
+
```
Comment (adrianherrera) [inline s2e_env/commands/code_coverage/line_info/__init__.py]: delete this line

File: s2e_env/commands/code_coverage/line_info/__init__.py [hunk 3]
```diff
@@ -50,4 +57,16 @@ def get_file_line_coverage(target_path, addr_counts):
...
+        logger.error('Could not read DWARF information from %s: %s', target_path, e)
+
+    try:
+        return jsoninfo.get_file_line_coverage(target_path, addr_counts)
+    except Exception as e:
+        logger.error('Could not get json line information from %s: %s\n'
+                     'If you try to get coverage for a Windows binary that has a PDB file,\n'
+                     'please use pdbparser.exe to get the json line information.', target_path, e)
+
+    raise CommandError('No usable line information available for %s' % (target_path))
```
Comment (adrianherrera) [inline s2e_env/commands/code_coverage/line_info/__init__.py]: target_path doesn't need () around it

File: s2e_env/commands/code_coverage/line_info/jsoninfo.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                current_file[line_number] = 0
+
+            if address in addr_counts:
+                current_file[line_number] += addr_counts[address]
+
+    return file_line_info
+
+
+def get_file_line_coverage(target_path, addr_counts):
+    target_path += '.lines'
```
Comment (adrianherrera) [inline s2e_env/commands/code_coverage/line_info/jsoninfo.py]: `target_path = '%s.lines' % target_path` :)

**Comment by adrianherrera** [comment]
This is really cool!

File: s2e_env/commands/code_coverage/line_info/jsoninfo.py (hunk 4 above)
Comment (vitalych) [reply s2e_env/commands/code_coverage/line_info/jsoninfo.py]: ok, but this is longer

File: s2e_env/commands/code_coverage/line_info/__init__.py (hunk 3 above)
Comment (vitalych) [reply s2e_env/commands/code_coverage/line_info/__init__.py]: ok

File: s2e_env/commands/code_coverage/line_info/dwarf.py (hunk 1 above)
Comment (vitalych) [reply s2e_env/commands/code_coverage/line_info/dwarf.py]: ok

File: s2e_env/commands/code_coverage/line_info/__init__.py (hunk 2 above)
Comment (vitalych) [reply s2e_env/commands/code_coverage/line_info/__init__.py]: I think it's ugly to have code right after a large comment. Is this mandated by pep8? Pycharm and pylint don't complain about it. They only complain if I add 2 lines:
image

Comment (adrianherrera) [reply s2e_env/commands/code_coverage/line_info/__init__.py]: Nah, it's not a PEP8 thing, I guess it's just a consistency thing.

**Pull request merged**